firmware/library/L1_Drivers/button_driver.hpp

```diff
@@ -1,10 +1,87 @@
 #pragma once
 
-class ButtonInterface {
+#include <cstdint>
+
+#include "L0_LowLevel/LPC40xx.h"
+#include "L1_Drivers/gpio.hpp"
+#include "L1_Drivers/pin_configure.hpp"
+
+class ButtonInterface
+{
+ public:
+         virtual void Initialize()                        = 0;
+         virtual bool Read()                              = 0;
+         virtual bool Released()                          = 0;
+         virtual bool Pressed()                           = 0;
+         virtual void InvertButtonSignal(bool enable_invert_signal = true)  = 0;
+};
+
+class ButtonDriver : public ButtonInterface
+{
  public:
-         virtual void Init()                = 0;
-         virtual bool Read()                = 0;
-         virtual bool Released()            = 0;
-         virtual bool Pressed()             = 0;
-         virtual void InvertButtonSignal(bool enable_invert_signal = true) = 0;
+       // Used for initialization of buttons
+       static constexpr uint8_t kFunction0 = 0b000;
+       static constexpr uint8_t kInput       = 0b0;
+
+       // Port 5, Port 4 (used for unit-testing)
+       static constexpr uint8_t kPortFive     = 5;
+       static constexpr uint8_t kPinFour      = 4;
+
+       constexpr ButtonDriver(uint8_t port_num, uint8_t pin_num)
+           : port_(port_num),
+             pin_(pin_num),
+             gpio_(gpio_pin_),
+             pincon_(pincon_pin_),
+             gpio_pin_(port_, pin_),
+             pincon_pin_(port_, pin_)
+       {
+       }
+
+       // Unit test constructor
+       constexpr ButtonDriver(const GpioInterface & gpio,
+                              const PinInterface & pincon)
+           : port_(kPortFive),
+             pin_(kPinFour),
+             gpio_(gpio),
+             pincon_(pincon),
+             gpio_pin_(port_, pin_),
+             pincon_pin_(Pin::CreateInactivePin())
+       {
+       }
+
+void Initialize() override {
+      pincon_pin_.SetMode(PinInterface::Mode::kInactive);
+      pincon_pin_.SetPinFunction(kFunction0);
+      gpio_pin_.SetAsInput();
+}
+
+bool Read() override {
+    return (gpio_pin_.ReadPin()) ? true : false;
+}
+
+bool Released() override {
+    return (Read()) ? true : false;
+}
+
+bool Pressed() override {
+    return (Read()) ? false : true;
+}
+
+void InvertButtonSignal(bool enable_invert_signal = true) override {
+  if (enable_invert_signal == false) {
```

Comment:
Comparison not needed. You can simply put the variable into the `SetAsActiveLow()` function.